### ISSUE TITLE
Stalker not working without AVX2 instructions

### ISSUE THREAD

**Issue by PinkiePieStyle** [frida/frida-gum#326]
OS: Microsoft Windows [Version 10.0.17763.316]
frida versions tested 12.2.20 - 12.4.0 + (debugbuild compiled from source #define FRIDA_VERSION "12.3.3")
tested with enable_jit() and without

C Portion - test.exe
(Compiled with Visual Studio 2017 15.9.6 compiler 19.16.27026.1 linker 14.16.27026.1 - default settings)

```
int main()
{
	while (true) {
		printf("Hello World!\n");
		Sleep(1000);
	}
}
```

JS Part
```
Process.setExceptionHandler(function(exception) {
    console.log(JSON.stringify(exception));
});

console.log(JSON.stringify(Process));

// from https://www.frida.re/docs/javascript-api/#stalker
Stalker.follow(Process.getCurrentThreadId(), {
    events: {
        call: true,
        ret: false,
        exec: false,
        block: false,
        compile: false
    },
    onCallSummary: function(summary) {
        console.log(JSON.stringify(summary));
    }
});
```

Output

```
* Trying to spawn test.exe
Hello World!
{"id":8532,"arch":"x64","platform":"windows","pageSize":4096,"pointerSize":8,"codeSigningPolicy":"optional"}

{"type":"illegal-instruction","address":"0x2045e2d103b","context":{"rip":"0x2045e2d103b","r15":"0x0","r14":"0x0","r13":"0x2045c633c80","r12":"0x2045c6a1d90","r11":"0x60c5416e74","r10":"0x2045c6b3690","r9":"0x5","r8":"0x1","rdi":"0x2045c6dc6d0","rsi":"0x0","rbp":"0x2045c78d160","rsp":"0x8e9e7ff120","rbx":"0x8e9e7ff420","rdx":"0x0","rcx":"0x2045c64abd0","rax":"0x8e9e7ff500","sp":"0x8e9e7ff120","pc":"0x2045e2d103b"},"nativeContext":"0x8e9e7fea80"}
```

Exception Handler changed to
```
Process.setExceptionHandler(function(exception) {
    console.log(JSON.stringify(exception));
    if(exception['type'] === 'illegal-instruction') {
        Stalker.unfollow(Process.getCurrentThreadId());
        Stalker.garbageCollect();
        return true;
    }
});
```

Output
```
* Trying to spawn test.exe
Hello World!
{"id":10660,"arch":"x64","platform":"windows","pageSize":4096,"pointerSize":8,"codeSigningPolicy":"optional"}

{"type":"illegal-instruction","address":"0x2621fff103b","context":{"rip":"0x2621fff103b","r15":"0x0","r14":"0x0","r13":"0x2621e3c3c80","r12":"0x2621e3aa4e0","r11":"0x60c5416e74","r10":"0x2621e42f0a0","r9":"0x5","r8":"0x1","rdi":"0x2621fcae980","rsi":"0x0","rbp":"0x2621e42dcb0","rsp":"0xf85d6ff610","rbx":"0xf85d6ff910","rdx":"0x0","rcx":"0x2621e4223b0","rax":"0xf85d6ff9f0","sp":"0xf85d6ff610","pc":"0x2621fff103b"},"nativeContext":"0xf85d6fef80"}
{"type":"access-violation","address":"0x2621fff103b","memory":{"operation":"execute","address":"0x2621fff103b"},"context":{"rip":"0x2621fff103b","r15":"0x0","r14":"0x0","r13":"0x2621e3c3c80","r12":"0x2621e3aa4e0","r11":"0x60c5416e74","r10":"0x2621e42f0a0","r9":"0x5","r8":"0x1","rdi":"0x2621fcae980","rsi":"0x0","rbp":"0x2621e42dcb0","rsp":"0xf85d6ff610","rbx":"0xf85d6ff910","rdx":"0x0","rcx":"0x2621e4223b0","rax":"0xf85d6ff9f0","sp":"0xf85d6ff610","pc":"0x2621fff103b"},"nativeContext":"0xf85d6fef80"}
```

**Comment by riverar** [comment]
Can't reproduce this. Do you have Core Isolation turned on, perhaps?

![image](https://user-images.githubusercontent.com/475132/60784169-bc241a80-a102-11e9-8377-23a5e34860dd.png)

**Comment by PinkiePieStyle** [comment]
Its disabled, updated Windows to latest ver 1903 (Build 18362.207) but same behavior
Tested on fresh Windows 1903 in vm with same results

**Comment by oleavr** [comment]
> Stalker.follow(Process.getCurrentThreadId(), {

This isn't really meaningful as you're trying to follow Frida's internal JavaScript thread. You either want to enumerate threads and follow one or more of them, or move this code to a hook (`onEnter` or `onLeave`).

**Comment by PinkiePieStyle** [comment]
Was playing around with fridasource and stalker.
Seems Stalker uses AVX2 instructions wich my CPU on the devpc doesnt support.
Error: vextracti128 xmmword ptr... illegal-instruction

**Comment by PinkiePieStyle** [comment]
Changed CPU on devmachine and
```
Stalker.follow(Process.getCurrentThreadId(), {
    events: {
        call: true,
        ret: false,
        exec: false,
        block: false,
        compile: false
    },
    onCallSummary: function(summary) {
        console.log(JSON.stringify(summary));
    }
});
```
works fine now
so problem was the missing AVX2 instructionset
thx for help anyway

**Comment by oleavr** [comment]
This was fixed in 23ff0cf6544d3bafb4861f3b6e78431e638def0c.
